tests/test_logging.py

```diff
@@ -188,8 +201,27 @@ def test_setup_logging_uses_env_var_overrides(tmp_path, dictConfigMock, monkeypa
     dictConfigMock.assert_called_once_with(expected_config)
 
 
+def test_setting_aliases_respected_for_logging_config(tmp_path: Path):
+    logging_config_content = """
+loggers:
+  prefect:
+    level: "${PREFECT_LOGGING_SERVER_LEVEL}"
+"""
+    config_file = tmp_path / "logging.yaml"
+    config_file.write_text(logging_config_content)
+
+    with temporary_settings(
+        {
+            PREFECT_LOGGING_SETTINGS_PATH: config_file,
+            PREFECT_SERVER_LOGGING_LEVEL: "INFO",
+        }
+    ):
+        config = setup_logging()
+        assert config["loggers"]["prefect"]["level"] == "INFO"
+
+
```

Comment:
@amgcc

the settings object has custom behavior including a getattr to fetch by env var name, which this Draft PR leverages

Reply:
this has now been updated to allow ourselves to dump the settings object with aliases, which makes the fix cleaner